extension/json/include/json_executors.hpp

```diff
@@ -29,6 +29,31 @@ struct JSONExecutors {
 		});
 	}
 
+	//! Single-argument JSON functions that (partially) exposes yyjson functionality
+	static void UnaryMutExecute(DataChunk &args, ExpressionState &state, Vector &result,
+	                            std::function<yyjson_mut_val *(yyjson_mut_val *, yyjson_alc *, Vector &)> fun) {
+		auto &lstate = JSONFunctionLocalState::ResetAndGet(state);
+		auto alc = lstate.json_allocator.GetYYAlc();
+
+		auto &inputs = args.data[0];
+		UnaryExecutor::Execute<string_t, string_t>(inputs, result, args.size(), [&](string_t input) {
+			// String to immutable yyjson document
+			auto doc = JSONCommon::ReadDocument(input, JSONCommon::READ_FLAG, alc);
+
+			// immutable document to mutable document
+			auto mut_doc = yyjson_doc_mut_copy(doc, alc);
+
+			// compute mutable result
+			auto new_val = fun(mut_doc->root, alc, result);
+
+			// Convert mutable value back to immutable document
+			doc = yyjson_mut_val_imut_copy(new_val, alc);
```

Comment:
Can't you avoid this copy by just calling `WriteVal<yyjson_mut_val>(doc->root, alc)` instead?

Reply:
thanks, done